editor/src/core/shared/uid-utils.ts

```diff
@@ -20,6 +20,67 @@ import { objectMap } from './object-utils'
 
 export const UtopiaIDPropertyPath = PP.create(['data-uid'])
 
+const atoz = [
+  'a',
+  'b',
+  'c',
+  'd',
+  'e',
+  'f',
+  'g',
+  'h',
+  'i',
+  'j',
+  'k',
+  'l',
+  'm',
+  'n',
+  'o',
+  'p',
+  'q',
+  'r',
+  's',
+  't',
+  'u',
+  'v',
+  'w',
+  'x',
+  'y',
+  'z',
+]
+
+// Assumes possibleStartingValue consists only of characters that are valid to begin with.
+export function generateConsistentUID(
+  existingIDs: Set<string>,
+  possibleStartingValue: string,
+): string {
+  if (possibleStartingValue.length >= 3) {
+    const maxSteps = Math.floor(possibleStartingValue.length / 3)
+    for (let step = 0; step < maxSteps; step++) {
+      const possibleUID = possibleStartingValue.substring(step * 3, (step + 1) * 3)
```

Comment:
So, thinking about it, we only went with the 3 character restriction on UIDs because they were previously included in the code. I wonder if it would make more sense to either remove this restriction entirely, or increase the allowed length to reduce the chance of collisions :thinking:

Reply:
It crossed my mind, but it seemed like a good idea to try and maintain the existing behaviour without changing too much and it makes it easier to eyeball things with shorter UIDs.

Reply:
I think increasing it to 4 or 5 characters would keep them debuggable, but make the collision space much larger